source/d3d12/runtime_d3d12.cpp

```diff
@@ -337,6 +375,32 @@ void reshade::d3d12::runtime_d3d12::on_present()
 	runtime::on_present();
 
 	_commandqueue->Signal(_fence[_swap_index].get(), ++_fence_value[_swap_index]);
+	clear_DSV_iter = 1;
+}
+
+void reshade::d3d12::runtime_d3d12::copy_depth_stencil(com_ptr<ID3D12Resource> src, com_ptr<ID3D12Resource> dest)
+{
+	com_ptr<ID3D12CommandQueue> copyQueue;
+	com_ptr<ID3D12CommandAllocator> copyAllocator;
+	com_ptr<ID3D12GraphicsCommandList> copyList;
+	HANDLE fenceEvent = nullptr;
+	com_ptr<ID3D12Fence> fence;
+
+	D3D12_COMMAND_QUEUE_DESC descCopyQueue = {};
+	descCopyQueue.Flags = D3D12_COMMAND_QUEUE_FLAG_NONE;
+	descCopyQueue.Type = D3D12_COMMAND_LIST_TYPE_DIRECT;
+	_device->CreateCommandQueue(&descCopyQueue, IID_PPV_ARGS(&copyQueue));
+	_device->CreateCommandAllocator(D3D12_COMMAND_LIST_TYPE_DIRECT, IID_PPV_ARGS(&copyAllocator));
+
+	if (FAILED(_device->CreateCommandList(0, D3D12_COMMAND_LIST_TYPE_COPY, copyAllocator.get(), nullptr, IID_PPV_ARGS(&copyList))))
```

Comment:
Don't create a new command queue and allocator here. The `runtime_d3d12` class already has the helper functions `create_command_list` and `execute_command_list` to do the work for you.

```c++
const com_ptr<ID3D12GraphicsCommandList> cmd_list = create_command_list();
transition_state(cmd_list, dest, D3D12_RESOURCE_STATE_COMMON, D3D12_RESOURCE_STATE_COPY_DEST);
cmd_list->CopyResource(dest.get(), src.get());
transition_state(cmd_list, dest, D3D12_RESOURCE_STATE_COPY_DEST, D3D12_RESOURCE_STATE_COMMON);
execute_command_list(cmd_list);
```

Reply:
Unfortunately, it does not fix random CTDs (Devil May Cry 5, SOTR). Maybe there is a need to add some fences